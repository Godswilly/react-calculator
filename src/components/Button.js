import React from 'react';
import PropTypes from 'prop-types';

export default function Button({ name }) {
  return <button type="button">{ name }</button>;
}

Button.propTypes = {
  name: PropTypes.string,
};

Button.defaultProps = {
  name: '',
};
