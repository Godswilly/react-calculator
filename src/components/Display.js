import React from 'react';
import PropTypes from 'prop-types';

export default function Display({ result, next }) {
  return (
    <div className="display">
      <h2>{ next || result || '0' }</h2>
    </div>
  );
}

Display.propTypes = {
  result: PropTypes.string,
  next: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
  next: null,
};
