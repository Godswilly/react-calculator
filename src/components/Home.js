import React from 'react';
import NavBar from './NavBar';

const Home = () => (
  <div>
    <NavBar />
    <h1>Welcome</h1>
  </div>
);

export default Home;
