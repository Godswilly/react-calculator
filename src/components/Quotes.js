import React from 'react';
import NavBar from './NavBar';

const Quotes = () => (
  <div>
    <NavBar />
    <h1>Welcome</h1>
    <p>
      Mathematics is not about numbers, equations, computations,
      or algorithms: it is about understanding. â€“William Paul Thurston
    </p>
  </div>
);

export default Quotes;
