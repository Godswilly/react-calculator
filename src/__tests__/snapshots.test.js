import React from 'react';
import renderer from 'react-test-renderer';
import Display from '../components/Display';
import Button from '../components/Button';
import Quotes from '../components/Quotes';
import Home from '../components/Home';

it('renders button correctly', () => {
  const tree = renderer.create(<Button name="AC" clickHandler={jest.fn()} />);
  expect(tree).toMatchSnapshot();
});

it('renders home page correctly', () => {
  const tree = renderer.create(<Home />);
  expect(tree).toMatchSnapshot();
});

it('renders display correctly', () => {
  const tree = renderer.create(<Display result="0" next="" />);
  expect(tree).toMatchSnapshot();
});

it('renders quotes page correctly', () => {
  const tree = renderer.create(<Quotes />);
  expect(tree).toMatchSnapshot();
});
