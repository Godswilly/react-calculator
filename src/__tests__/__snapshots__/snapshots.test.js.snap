// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button correctly 1`] = `
<button
  onClick={[MockFunction]}
  type="button"
>
  AC
</button>
`;

exports[`renders display correctly 1`] = `
<div
  className="display"
>
  <h2>
    0
  </h2>
</div>
`;

exports[`renders home page 1`] = `
<div>
  <nav
    className="navbar"
  >
    <h1>
      Math Magicians
    </h1>
    <ul
      className="navbar-menu"
    >
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/App"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/Quotes"
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
  <h2>
    Welcome
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
  </p>
</div>
`;

exports[`renders quotes page 1`] = `
<div>
  <nav
    className="navbar"
  >
    <h1>
      Math Magicians
    </h1>
    <ul
      className="navbar-menu"
    >
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/App"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/Quotes"
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
  <h2>
    Welcome
  </h2>
  <p>
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. â€“William Paul Thurston
  </p>
</div>
`;
